src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Identification/ExceptionIdentifierCache.cs

```diff
@@ -79,6 +85,14 @@ public ulong GetOrAdd(ExceptionIdentifier exceptionId)
         /// </returns>
         public ulong GetOrAdd(MethodBase method)
         {
+            // Dynamic methods do not have method handles and can be detected by checking if their declaring type is null
+            // Do not attempt to create an identifier for these and skip them.
+            // CONSIDER: Can an artificial identifier be created that does not collide with the algorithm used in GetId(MethodBase)?
+            if (null == method.DeclaringType)
```

Comment:
Are there situations where declaring type is null but method id is not null? In the profiler this is possible (I.e. FunctionId is not null, but the method is not associated with a type)

Reply:
Another interesting scenario is arrays. I.e. throw an exception from one of the methods on array instance. Does declaring type resolve to Array or somethign else.

Reply:
I reworked this part to allow handling of dynamic methods and catch exceptions appropriately for API calls that will fail for them; ideally I would be able to easily identify a dynamic method but the only way I found to do that was to check the runtime type of `method` and see if it's a RTDynamicMethod (this type is internal to System.Private.CoreLib), but I decided against doing that type of check since we don't care that it is dynamic and just want to get as best information out of it as possible. So now we only using `DeclaringType` to get the owner type of the method.

Reply:
Arrays work fine; the `DeclaringType` is `System.Array`.